utils/estimated_histogram.hh

```diff
@@ -47,13 +47,302 @@
 #include <chrono>
 #include <seastar/core/metrics_types.hh>
 #include <seastar/core/print.hh>
-
-#include <seastar/core/print.hh>
-
 #include "seastarx.hh"
+#include <seastar/core/bitops.hh>
+#include <limits>
+#include <array>
 
 namespace utils {
 
+/**
+ * This is a pseudo floating point implementation of an estimated histogram.
```

Comment:
The implementation is similar to pseudo-floating-point implementation.

Giving a range: power of 2 growing sub-ranges (ie. exponential with a coefficient of 2)
Each of those sub-ranges is split linearly.

Reply:
Oh! I think I understood now why you called it "pseudo floating point": It has the exponent part, and inside each "range" defined by the exponent, you have the mantissa part - which addresses the individual points linearly. 
I think that this is a very interesting analogy. But I think it would not be intuitively clear to readers (it wasn't to me), so it shouldn't be part of the commit message and the first line of the comment. It would be nice, indeed, to have it in a comment or the commit message, but not so prominently.
Or so I think.

Reply:
@nyh Thanks for all the other comments as well, it help me see all the places that it was not clear for an external reader.
The next version will be submitted soon where I address it